ee/billing/quota_limiting.py

```diff
@@ -0,0 +1,31 @@
+from datetime import timedelta
+from enum import Enum
+from typing import List, Mapping
+
+from django.utils import timezone
+
+from posthog.cache_utils import cache_for
+from posthog.redis import get_client
+
+RATE_LIMITER_CACHE_KEY = "@posthog/quota-limits/"
+
+
+class QuotaResource(Enum):
+    EVENTS = "events"
+    RECORDINGS = "recordings"
+
+
+def replace_limited_teams(resource: QuotaResource, tokens: Mapping[str, int]) -> None:
+    pipe = get_client().pipeline()
+    pipe.zremrangebyscore(f"{RATE_LIMITER_CACHE_KEY}{resource.value}", "-inf", "+inf")
```

Comment:
Because it is so cheap anyway that the difference is negligible 🤷 
We want the time that the rate limit is valid until so that we can stop limiting as soon as the billing period ends. 

Regarding your main comment - as far as I understood from pipeline team, the plan is to get rid of Postgres as a part of the capture flow which is why we opted for Redis.

Reply:
Also to be clear - this is a solution _for now_. We have been trying to get this in for over a month but keep running in circles. Totally happy for this to get completely replaced by some super clever mechanism but at the moment we just need something that works 😅

Reply:
Works!

Reply:
fwiw, instead of remrangebyscore, you can also just delete the key, not sure if it works better (but I'd expect yes).

My only worry is these kind of operations block and can be detrimental to anything else that depends on redis (like decide) - I found this out the hard way, trying to list existing keys given a team, and ended up increasing decide latency from 200ms to 1second because this was blocking, for the duration of the operation 🤯

Reply:
Yeah, consider everything I say as non-blocking and do what you got to do!

Reply:
👍 Yeah all of this is _somewhat_ thought out in the fact that the new memory cache logic will only go to redis in the background. It _should_ mean that worst case we don't rate limit people that we should if for whatever reason redis is unavailable and if it is just being slow - no biggy as it is all happening in the background

Reply:
> fwiw, instead of remrangebyscore, you can also just delete the key, not sure if it works better (but I'd expect yes).

I don't know why I didn't do this 😅